src/main/webapp/app/modules/conflicts/single/components/single-location-details.tsx

```diff
@@ -0,0 +1,116 @@
+import React from 'react';
+import { Col, Row, Button } from 'reactstrap';
+import '../single-record-view.scss';
+import { Translate } from 'react-jhipster';
+import { connect } from 'react-redux';
+import { IActivity } from 'app/shared/model/activity.model';
+import { ILocation } from 'app/shared/model/location.model';
+import { PhysicalAddressDetails } from './physical-address-details';
+import { PostalAddressDetails } from './postal-address-details';
+import { OpeningHoursDetails } from './opening-hours-details';
+import { IPhysicalAddress } from 'app/shared/model/physical-address.model';
+import { IPostalAddress } from 'app/shared/model/postal-address.model';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import { IOpeningHours } from 'app/shared/model/opening-hours.model';
+import { AdditionalDetails } from './additional-details';
+
+export interface ISingleLocationDetailsProp extends StateProps, DispatchProps {
+  activity: IActivity;
+  location: ILocation;
+  physicalAddress: IPhysicalAddress;
+  postalAddress: IPostalAddress;
+  locationsCount: string;
+  changeRecord: any;
+  isOnlyOne: boolean;
+  hours: IOpeningHours[];
+}
+
+export interface ISingleLocationDetailsState {
+  isAreaOpen: boolean;
+}
+
+export class SingleLocationDetails extends React.Component<ISingleLocationDetailsProp, ISingleLocationDetailsState> {
+  state: ISingleLocationDetailsState = {
+    isAreaOpen: false
+  };
+
+  toggleAreaOpen = () => {
+    this.setState({
+      isAreaOpen: !this.state.isAreaOpen
+    });
+  };
+
+  getTextField = (location, fieldName) => {
+    return {
+      type: 'text',
+      fieldName: fieldName,
+      defaultValue: location[fieldName]
+    };
+  };
+
+  render() {
+    const { location, physicalAddress, postalAddress, isOnlyOne, hours } = this.props;
+    const customHeader = (
+      <h4 className="title">
+        <div className="collapseBtn" onClick={this.toggleAreaOpen}>
+          <div className="collapseIcon">
+            <FontAwesomeIcon size="xs" icon={this.state.isAreaOpen ? 'angle-up' : 'angle-down'} />
+          </div>
+          <Translate contentKey="singleRecordView.details.titleLocations" /> {this.props.locationsCount}
+        </div>
+        {isOnlyOne ? null : (
+          <Button className="primary" onClick={this.props.changeRecord}>
+            <Translate contentKey="singleRecordView.details.seeAnotherRecord" />
+          </Button>
+        )}
+      </h4>
```

Comment:
Can you take h4 to another component

Reply:
Also, this header differs from the two others